core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java

```diff
@@ -31,6 +31,11 @@
  *
  * Prefer implementing this interface over implementing {@link Streamable} where possible. Lots of code depends on {@linkplain Streamable}
  * so this isn't always possible.
+ *
+ * The fact that this interface extends {@link StreamableReader} should be consider vestigial. Instead of using its
```

Comment:
vesitigial: that made me go straight to the english dictionary... maybe silly question, but why do we extend StreamableReader here at all? is it something that we will get rid of at some point but we cannot do it now?